src/command/graph/push.rs

```diff
@@ -43,11 +44,21 @@ impl Push {
 
         tracing::debug!("Schema Document to push:\n{}", &schema_document);
 
+        let git = GitContext::new();
+        tracing::debug!("Git Context: {:?}", git);
+
         let push_response = push::run(
             push::push_schema_mutation::Variables {
                 graph_id: self.graph.name.clone(),
                 variant: self.graph.variant.clone(),
                 schema_document: Some(schema_document),
+                git_context: Some(push::push_schema_mutation::GitContextInput {
+                    branch: git.branch,
+                    committer: git.committer,
+                    commit: git.commit,
+                    message: git.message,
+                    remote_url: git.remote_url,
+                })
```

Comment:
i wonder if @lrlna knows of any special tricks to make this conversion a bit less verbose since all of the field names are the exact same

Reply:
yes, there is a way! Because `GitContextInput` and `GitContext` are two different types, we need to implement an [`Into` trait](https://doc.rust-lang.org/std/convert/trait.Into.html) for `GitContext`. Something like this:

```rust
struct GitContext;
impl Into<push::push_schema_mutation::GitContextInput> for GitContext {
    fn into(self) -> GitContext {
        GitContextInput {
          branch: self.branch, // all other fields   
       }
    }
}
```

You will have to do this for `check::` version of `GitContextInput` as well. And then here, instead of doing this iteration you can just go `git.into()`, which is very sleek indeed :). (Fun fact, if they _were_ the same type, we could have just done `...`, but one can only dream, eh)

Reply:
i knew it! this sounds perfect 😄